tests/core/integrations/test_diarization.py

```
@@ -0,0 +1,654 @@
+# diarize() — pyannote not installed
+# ---------------------------------------------------------------------------
+
+
+def test_diarize_raises_when_pyannote_not_installed() -> None:
+    """diarize() raises DiarizationError when Pipeline is None."""
+    result = _sample_result()
+    with patch(
+        "insanely_fast_whisper_rocm.core.integrations.diarization.Pipeline",
+        None,
+    ):
+        with pytest.raises(DiarizationError, match="pyannote.audio is not installed"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)